### PR TITLE
Remove default icons and refactor icon-lookup

### PR DESCRIPTION
**Pull request by bynect**
As requested by #1364 and #1358. I also refactored the search function to provide more informative logging

### CODE REVIEW

**Comment by bynect** [comment]
some  cleanup

File: test/icon-lookup.c [hunk 1]
```diff
@@ -55,20 +46,6 @@ TEST test_find_icon(void)
         PASS();
 }
 
-TEST test_find_path(void)
```
Comment (fwsmit) [inline test/icon-lookup.c:58]: Why did you remove these tests? Did anything in the search behavior change other than the default icons? If so that should also be documented and discussed

Comment (bynect) [reply test/icon-lookup.c:58]: find_icon_path and get_path_from_icon had basically half of the code duplicate. so I removed some code from the first since in the first place it is used only by get_path_from_icon

Comment (bynect) [reply test/icon-lookup.c:58]: the absolute path and uri handling were the exact same

Comment (fwsmit) [reply test/icon-lookup.c:58]: Ah okay. Makes sense

**Comment by bynect** [comment]
@fwsmit does it looks good? can I merge?

**Comment by fwsmit** [comment]
> @fwsmit does it looks good? can I merge?

Yeah, looks good. You can go ahead and merge

**Review by fwsmit — APPROVED** [review]

**Comment by bynect** [comment]
> > @fwsmit does it looks good? can I merge?
> 
> Yeah, looks good. You can go ahead and merge

perfect 👍🏻
